services/containers/service.go

```diff
@@ -120,6 +134,15 @@ func (s *Service) Update(ctx context.Context, req *api.UpdateContainerRequest) (
 		var fieldpaths []string
 		if req.UpdateMask != nil && len(req.UpdateMask.Paths) > 0 {
 			for _, path := range req.UpdateMask.Paths {
+				if strings.HasPrefix(path, "labels.") {
```

Comment:
Actually, this logic is already handled by the store itself, so you should move the validation there. I believe that will be in the `metadata` package.